Assets/Scripts/HighlightController.cs

```diff
@@ -0,0 +1,46 @@
+using UnityEngine;
+
+public class HighlightController : MonoBehaviour
+{
+    private Camera mainCamera;
+    private Highlight currentHighlight;
+
+    private void Start()
+    {
+        mainCamera = Camera.main;
+    }
+
+    private void FixedUpdate()
+    {
+        LookForHighlightableObject();
+    }
+
+    /// <summary>
+    /// Highlight objects with highlight script when looking at them
+    /// </summary>
+    private void LookForHighlightableObject()
+    {
+        Highlight foundHighlight = null;
+        RaycastHit hit;
+        if (Physics.Raycast(mainCamera.transform.position, mainCamera.transform.forward, out hit, Mathf.Infinity))
```

Comment:
Instead of Infinite recast a variable for the possession range would be better

Reply:
I added a max range variable